inlang/source-code/website/src/pages/blog/@id/index.page.server.tsx

```diff
@@ -1,80 +1,57 @@
-import type { OnBeforeRender } from "#src/renderer/types.js"
-import type { PageProps } from "./index.page.jsx"
-import { parseMarkdown } from "#src/services/markdown/index.js"
+import { convert } from "@inlang/markdown"
 import { RenderErrorPage } from "vite-plugin-ssr/RenderErrorPage"
 import fs from "node:fs/promises"
-import { BlogFrontmatterSchema } from "./frontmatterSchema.js"
 import tableOfContents from "../../../../../../blog/tableOfContents.json"
+import type { OnBeforeRender } from "#src/renderer/types.js"
+import type { PageProps } from "./index.page.jsx"
 
-/**
- * The root of the repository.
- *
- * Makes it possible to use absolute paths for rendering markdown.
- */
 const repositoryRoot = new URL("../../../../../../", import.meta.url)
 
-/**
- * the table of contents without the html for each document
- * saving bandwith and speeding up the site)
- */
-export type ProcessedTableOfContents = Record<
+export type GeneratedTableOfContents = Record<
 	string,
-	Awaited<ReturnType<typeof parseMarkdown>>["frontmatter"]
+	{
+		title: string
+		description: string
+		href: string
+	}
 >
 
-/**
- * The index of documentation markdown files. The href's acts as id.
- *
- * @example
- * 	{
- * 		"/documentation/intro": document,
- * 	}
- */
-const index: Record<string, Awaited<ReturnType<typeof parseMarkdown>>> = {}
-/**
- * the table of contents without the html for each document
- * saving bandwith and speeding up the site)
- */
-const processedTableOfContents: ProcessedTableOfContents = {}
+const index: Record<string, Awaited<ReturnType<any>>> = {}
+const generatedTableOfContents = {} as GeneratedTableOfContents
 
 await generateIndexAndTableOfContents()
 
-// should only run server side
 export const onBeforeRender: OnBeforeRender<PageProps> = async (pageContext) => {
-	// dirty way to get reload of markdown (not hot reload though)
 	if (import.meta.env.DEV) {
 		await generateIndexAndTableOfContents()
 	}
 	if (
-		!Object.keys(index).includes(pageContext.urlPathname) &&
+		!Object.keys(index).includes(pageContext.urlPathname.replace("/blog/", "")) &&
 		pageContext.urlPathname !== "/blog"
 	) {
 		throw RenderErrorPage({ pageContext: { is404: true } })
 	}
+
 	return {
 		pageContext: {
 			pageProps: {
-				markdown: index[pageContext.urlPathname]!,
-				processedTableOfContents: processedTableOfContents,
+				markdown: index[pageContext.urlPathname.replace("/blog/", "")]!,
+				meta: generatedTableOfContents[pageContext.urlPathname.replace("/blog/", "")]!,
```

Comment:
What is meta for?

Reply:
SEO

Reply:
But you get that from the description don't you? If so, name it description :)